roles/collect/tasks/main.yml

```diff
@@ -76,12 +65,17 @@
 # this appends the string inside the brackets to the 'api_urls' list.
 - name: assemble compiled URL facts, append to list.
   set_fact:
-    api_urls: "{{ api_urls + [api_host+'/'+api_path+item] }}"
+    api_urls: "{{ api_urls + [api_path+item] }}"
   with_items: "{{ api_params }}"
 
+# Use a cluster specific path
+- name: Set download_path
+  set_fact:
+    temp_download_path: "{{ collect_download_path }}/{{ ocp_config['ocp_cluster_id'] }}/korekuta_collect"
+
 - name: Create temp dir for downloaded files
   file:
-    path: '{{ collect_download_path }}'
+    path: '{{ temp_download_path }}'
```

Comment:
Please pre-pend `collect_` to global variable names to namespace the role's variables and avoid variable name collisions with other roles.